daemon/src/taker.rs

```diff
@@ -23,19 +27,55 @@ mod wire;
 #[database("taker")]
 pub struct Db(sqlx::SqlitePool);
 
+#[derive(Clap)]
+struct Opts {
+    /// The IP address of the taker to connect to.
+    #[clap(long, default_value = "127.0.0.1:9999")]
+    taker: SocketAddr,
+
+    /// The port to listen on for the HTTP API.
+    #[clap(long, default_value = "8000")]
+    http_port: u16,
+
+    /// URL to the electrum backend to use for the wallet.
+    #[clap(long, default_value = "ssl://electrum.blockstream.info:60002")]
+    electrum: String,
+
+    /// Where to permanently store data, defaults to the current working directory.
+    #[clap(long)]
+    data_dir: Option<PathBuf>,
+
+    /// Generate a seed file within the data directory.
+    #[clap(long)]
+    generate_seed: bool,
+}
+
 #[rocket::main]
 async fn main() -> Result<()> {
-    let client =
-        bdk::electrum_client::Client::new("ssl://electrum.blockstream.info:60002").unwrap();
+    let opts = Opts::parse();
+
+    let data_dir = opts
+        .data_dir
+        .unwrap_or_else(|| std::env::current_dir().expect("unable to get cwd"));
+
+    if !data_dir.exists() {
+        tokio::fs::create_dir_all(&data_dir).await?;
+    }
+
+    let seed = Seed::initialize(&data_dir.join("seed"), opts.generate_seed).await?;
+
+    let client = bdk::electrum_client::Client::new(&opts.electrum).unwrap();
 
     // TODO: Replace with sqlite once https://github.com/bitcoindevkit/bdk/pull/376 is merged.
-    let db = bdk::sled::open("/tmp/taker.db")?;
+    let db = bdk::sled::open(data_dir.join("wallet_db"))?;
```

Comment:
I believe you should maintain unique wallet db names for both sides - there's no enforcement of specifying different data dirs for maker and the taker at the moment, and after this changeset you might end up trying to open the same wallet db on both ends.
```suggestion
    let db = bdk::sled::open(data_dir.join("taker_wallet_db"))?;
```

Reply:
Fair enough, also need to change the name of the seed file then!